File: app/src/main/java/com/platdmit/simplecloudmanager/data/api/rest/RestServer.java
package com.platdmit.simplecloudmanager.data.api.rest;

import com.platdmit.simplecloudmanager.data.api.models.ApiRequestBody;

import retrofit2.Call;
import retrofit2.http.DELETE;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface RestServer {

    @GET("vps")
    Call<ApiRequestBody> getServers();

    @GET("vps/{id}")
    Call<ApiRequestBody> getServer(@Path("id") long id);

    @GET("vps/{id}/actions")
    Call<ApiRequestBody> getServerActions(@Path("id") long id);

    @GET("vps/{id}/statistics")
    Call<ApiRequestBody> getServerStatistics(@Path("id") long id);

    @GET("vps/{id}/loadAverage")
    Call<ApiRequestBody> getServerLoadAverage(@Path("id") long id);

    @GET("vps/{id}/backups")
    Call<ApiRequestBody> getServerBackups(@Path("id") long id);

    @DELETE("vps/{id}")
    void dellServer(@Path("id") long id);
}

File: data/src/main/java/com/platdmit/data/retrofit/rest/RestServer.kt
package com.platdmit.data.retrofit.rest

import com.platdmit.data.retrofit.models.ApiRequestBody
import retrofit2.Call
import retrofit2.http.DELETE
import retrofit2.http.GET
import retrofit2.http.Path

interface RestServer {
    @GET("vps")
    fun getServers(): Call<ApiRequestBody>

    @GET("vps/{id}")
    fun getServer(@Path("id") id: Long): Call<ApiRequestBody>

    @GET("vps/{id}/actions")
    fun getServerActions(@Path("id") id: Long): Call<ApiRequestBody>

    @GET("vps/{id}/statistics")
    fun getServerStatistics(@Path("id") id: Long): Call<ApiRequestBody>

    @GET("vps/{id}/loadAverage")
    fun getServerLoadAverage(@Path("id") id: Long): Call<ApiRequestBody>

    @GET("vps/{id}/backups")
    fun getServerBackups(@Path("id") id: Long): Call<ApiRequestBody>

    @DELETE("vps/{id}")
    fun dellServer(@Path("id") id: Long)
}